### DISCUSSION TITLE
Enable pdf attachments to be sent to external APIs

### DISCUSSION THREAD

**Discussion by legaltextai** [open-webui/open-webui#5485] [General]
At the moment, any message with pdf attachment goes via RAG pipeline. 
I would like to be able to use external API to send the file to. 
Please make the RAG pipeline optional when user attaches a pdf. 
Thank you for your absolutely incredible product!

**Comment by tjbck** [comment]
Hope https://github.com/open-webui/open-webui/discussions/3687#discussioncomment-10016728 helps!

**Comment by legaltextai** [comment]
Thank you!  Managed to get the file sent to external API (the external API takes a badly scanned pdf and converts it to images). But, the problem is if the pdf is readable, it still kicks in the default extraction of text. How to disable it and have all files sent to external API? 
Thank you

**Comment by antonioliveira84** [comment]
I don't know if this is still useful:

PDF_EXTRACT_IMAGES
Type: bool
Default: False
Description: Extracts images from PDFs using OCR when loading documents.
Persistence: This environment variable is a PersistentConfig variable.

https://docs.openwebui.com/getting-started/env-configuration/#:~:text=PDF_EXTRACT_IMAGES%E2%80%8B,a%20PersistentConfig%20variable.

**Reply by rgaricano** [reply]
Yes, is the default OCR set for uploaded pdfs,
some engines use it, others have it owns configs and don't use it.
### (PDF_EXTRACT_IMAGES could be extended to other images on differents docs types, or to uploaded images!!!)
